src/Attribute.cpp

```diff
@@ -0,0 +1,57 @@
+#if HAVE_CONFIG_H
+#  include <config.h>
+#endif
+
+#ifdef WIN32
+#prattr_genma warning(disable : 4786)   /* Disable annoying warning messattr_genes */
+#endif
+
+#include "random.h"
+#include "OutputMgr.h"
+#include "Attribute.h"
+
+AttributeGenerator::AttributeGenerator()
+	: attr_emitted(false)
+{
+}
+
+
+Attribute::Attribute(string name, int prob)
+	:  attribute(name), attribute_probability(prob)
+{
+}
+
+void
+Attribute::OutputAttribute(std::ostream &out, string option, AttributeGenerator &attr_gen)
```

Comment:
Done. Passed bool variable `attr_emitted` instead of object.

Reply:
No, `OutputAttribute` should not be passed such an argument. Let the Generator print `"__attribute__ (("` and corresponding ending.

Reply:
But generator does not know how many attributes will pass `rnd_flipcoin()` probability. It can only print individual attributes like `__attribute__((hot)) __attribute__((cold))` in current state if we dont pass `object` or `attr_emitted`.

Reply:
What about `OutputAttribute` function indicating whether it adds and attribute or not?

Reply:
Like returning a bool variable from `OutputAttribute` whether an attribute have passed `rnd_flipcoin()` in `OutputAttributes()`?

Reply:
Well, I would prefer `OutputAttribute` returns a string which can then be printed to the stream by `AttributeGenerator`.

Reply:
Done. Please check it.